[ModuleInterface] Include the raw values of @objc enums

jrose-apple opened the pull request:
...since they're part of the run-time representation. Not having this meant that someone compiling against an interface would miscompile uses of `@objc` enums defined in that interface!

rdar://problem/50410541

jrose-apple commented:
@swift-ci Please test

harlanhaskins reviewed (approved):
Thanks for fixing this!

jrose-apple commented:
@swift-ci Please test

jrose-apple committed:
[ModuleInterface] Include the raw values of @objc enums

...since they're part of the run-time representation. Not having this
meant that someone compiling against an interface would miscompile
uses of @objc enums defined in that interface!

rdar://problem/50410541

jrose-apple commented:
@swift-ci Please test
